examples/solids/index.rst

```diff
@@ -0,0 +1,348 @@
+replaced with a boundary condition.
+
+The stress-strain relationship (constitutive law) is given by:
+
+.. math::
+   :label: linear-stress-strain
+
+   \boldsymbol{\sigma} = \mathsf{C} \boldsymbol{\epsilon},
+
+where 
+
+.. math::
+
+   \boldsymbol{\epsilon} = \dfrac{1}{2}\left(\nabla \boldsymbol{u} + \nabla \boldsymbol{u}^T \right)
+
+is the (symmetric) strain tensor.
+For notational convenience, we suppose the symmetric second order tensors :math:`\bm \sigma` and :math:`\bm \epsilon` are expressed as vectors of length 6 using `Voigt notation <https://en.wikipedia.org/wiki/Voigt_notation>`_, in which case the fourth order material stiffness tensor :math:`\mathsf C` can be represented as a :math:`6\times 6` symmetric matrix
+
+.. math::
+
+   \mathsf C = \dfrac{E}{(1+\nu)(1-2\nu)}
+   \begin{bmatrix}
+        1-\nu & \nu & \nu & & & \\
+          \nu & 1 - \nu & \nu & & & \\
+          \nu & \nu &  1 - \nu & & & \\
+          & & & \dfrac{1 - 2\nu}{2} & & \\    
+         & & & &\dfrac{1 - 2\nu}{2} & \\
+         & & & & & \dfrac{1 - 2\nu}{2} \\   
+   \end{bmatrix},
+
+where :math:`E` is the Young’s modulus and :math:`\nu` is the Poisson’s ratio.
+An alternate formulation is in terms of the Lamé parameters,
+
+.. math::
+   \begin{aligned}
+   \lambda &= \frac{E \nu}{(1 + \nu)(1 - 2 \nu)} \\
+   \mu &= \frac{E}{2(1 + \nu)}
+   \end{aligned}
+
+in which case the constitutive equation :math:numref:`linear-stress-strain` may be written
+
+.. math::
+   \bm\sigma = \lambda (\operatorname{trace} \bm\epsilon) \bm I_3 + 2 \mu \bm\epsilon,
+
+where :math:`\bm I_3` is the :math:`3 \times 3` identity matrix.
+Alternatively, the stiffness tensor becomes
+
+.. math::
+
+   \mathsf C = \begin{bmatrix}
+   \lambda + 2\mu & \lambda & \lambda & & & \\
+   \lambda & \lambda + 2\mu & \lambda & & & \\
+   \lambda & \lambda & \lambda + 2\mu & & & \\
+   & & & \mu & & \\
+   & & & & \mu & \\
+   & & & & & \mu
+   \end{bmatrix}.
+
+Note that the incompressible limit :math:`\nu \to \frac 1 2` causes :math:`\lambda \to \infty`, and thus :math:`\mathsf C` becomes singular.
+
+.. _problem-hyper-small-strain:
+
+Hyperelasticity at Small Strain
+----------------------------------------
+
+The strong form of the static balance linear momentum at small strain for the three-dimensional neo-Hookean Hyperelasticity problem is given by
+:cite:`hughes2012finite`:
+
+.. math::
+
+   \nabla \cdot \boldsymbol{\sigma} + \boldsymbol{g} = \boldsymbol{0} 
+
+Integrating by parts on the divergence term, we arrive at the weak form:
+
+.. math::
+
+   \int_{\Omega}{ \nabla \boldsymbol{v} \colon \boldsymbol{\sigma}} dV - \int_{d\Omega}{\boldsymbol{v} \cdot \left(\boldsymbol{\sigma}_t \cdot \hat{\boldsymbol{n}}\right)} dS + \int_{\Omega}{\boldsymbol{v} \cdot \boldsymbol{g}} dV = 0
+
+where :math:`\boldsymbol{\sigma}_t \cdot \hat{\boldsymbol{n}}` is typically replaced with a boundary condition.
+
+The small strain version of a Neo-Hookean hyperelasticity material is given as
```

Comment:
Even the first time strain is introduced we should mention that it is small/infinitesimal strain, otherwise it sounds like peculiar to this Neo-Hookean constitutive relationship, and it is not.

Reply:
I stand by my initial recommendation that we introduce the fully nonlinear case first and linearize parts geometry and materials to derive the linear case as a simplified model for testing purposes. (I don't think there will be a big performance difference for a good implementation. Linearization might reduce 2 Newton iterations to 1 and make a modest difference in operator application time.) I don't know if @jeremylt and @ArashMehraban affirmatively didn't like that suggestion or just had this text and didn't want to spend time refactoring it.

Reply:
It seemed like @ArashMehraban took the lead on the documentation, so I hadn't jumped in to avoid too many cooks. I have no strong opinions here.

Reply:
Yep, and he needs practice writing about this stuff for his paper. I'm happy for it to be in his hands. I (re)wrote the hyperelastic/constitutive modeling section because I felt like it was the most efficient way to communicate about that essential core structure, but I'd like him to help make the whole document fit together. And thanks @valeriabarra for your review and edits.